src/ica_api.c

```diff
@@ -3711,6 +3711,30 @@ unsigned int ica_aes_gcm_initialize(const unsigned char *iv,
 #endif /* NO_CPACF */
 }
 
+unsigned int ica_aes_gcm_initialize_fips(unsigned char *iv,
+		unsigned int iv_length, unsigned char *key, unsigned int key_length,
+		unsigned char *icb, unsigned char *ucb, unsigned char *subkey,
+		unsigned int direction)
+{
+	int rc;
+
+#ifdef ICA_FIPS
+	if (fips >> 1)
+		return EACCES;
+	if (fips & ICA_FIPS_MODE) {
+		if (iv_length < GCM_RECOMMENDED_IV_LENGTH)
+			return EPERM;
+	}
+#endif
+
+	rc = RAND_bytes(iv, iv_length);
```

Comment:
Check for iv being NULL and return EINVAL if so? Or does RAND_bytes do this already? But better not rely on argument checking of an external library function.